src/main/java/org/jabref/model/entry/identifier/SSRN.java

```diff
@@ -0,0 +1,78 @@
+package org.jabref.model.entry.identifier;
+
+import java.net.URI;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents an SSRN article, identified by its abstract ID.
+ */
+public class SSRN extends EprintIdentifier {
+
+    private static final String SSRN_URL_REGEX = "(https?://)?(papers\\.)?ssrn\\.com/(sol3/papers.cfm\\?)?abstract(_id)?=(?<id>\\d+)";
+    private static final Pattern SSRN_URL_FULL_MATCH = Pattern.compile("^" + SSRN_URL_REGEX + "$", Pattern.CASE_INSENSITIVE);
+    private static final Pattern SSRN_URL_MATCH = Pattern.compile(SSRN_URL_REGEX, Pattern.MULTILINE | Pattern.CASE_INSENSITIVE);
+
+    private final int abstractId;
+
+    /**
+     * Tries to parse an SSRN identifier
+     *
+     * @param string Either a number or a SSRN url that has the abstract ID in it
+     * @throws NullPointerException If you pass a null to the constructor
+     * @throws IllegalArgumentException Invalid string passed to the constructor
+     */
+    public SSRN(String string) {
+        Objects.requireNonNull(string);
+        string = string.trim();
+
+        if (string.chars().allMatch(Character::isDigit)) {
+            abstractId = Integer.parseInt(string);
+            return;
+        }
+
+        Matcher matcher = SSRN_URL_FULL_MATCH.matcher(string);
+        if (matcher.find()) {
+            abstractId = Integer.parseInt(matcher.group("id"));
+            return;
+        }
+
+        throw new IllegalArgumentException(string + " is not a valid SSRN identifier");
+    }
+
+    public SSRN(int abstractId) {
```

Comment:
This is for testing only, isn't it?

Reply:
It is also used inside of parse.

(I will work on the other comments tomorrow morning)